### PR TITLE
[ACS-6019] [ACS-6021] [ACS-6023] Replaced request with node-fetch

### PR DESCRIPTION
**Pull request by swapnil-verma-gl**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Codebase was using depracated request library

**What is the new behaviour?**
Replaced request library with node-fetch library

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by AleksanderSklorz — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -130,7 +131,6 @@
     "@types/minimatch": "^3.0.3",
     "@types/node": "18.0.0",
     "@types/pdfjs-dist": "^2.10.378",
-    "@types/request": "^2.48.9",
```
Comment (MichalKinas) [inline package.json:133]: You removed `request types` but the `request` library is still there so I guess you should remove it as well. Also I think you should run `npm i` afterwards to regenerate `package-lock.json`

File: lib/cli/scripts/init-aae-env.ts [hunk 2]
```diff
@@ -665,18 +665,29 @@ function findFailingApps(deployedApps: any[]): any[] {
  * @param name name
  */
 async function getFileFromRemote(url: string, name: string) {
```
Comment (MichalKinas) [inline lib/cli/scripts/init-aae-env.ts:667]: Can we add return type for this method?

Comment (DenysVuika) [reply lib/cli/scripts/init-aae-env.ts:667]: I guess it's `Promise<void>`

**Comment by swapnil-verma-gl** [comment]
Just FYI, since the adf-cli project is currently using CommonJS modules and not ESModules, I had to downgrade node-fetch version from v3.3.2 to v2.7.0. The reason for the change is node-fetch from v3 onwards is ESModule compatible only. For CommonJS modules, they [recommend](https://github.com/node-fetch/node-fetch#commonjs) sticking to v2

**Pull request merged**